src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext, useRef, useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../contexts/TodosProvider';
+import { TodoAction } from '../../types/TodoAction';
+
+type Props = {
+  todo: Todo,
+};
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+function customDebounce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (...args: any) => {
+    if (timerId) {
+      window.clearTimeout(timerId);
+      callback(...args);
+      timerId = 0;
+    }
+
+    timerId = window.setTimeout(() => {
+      window.clearTimeout(timerId);
+      timerId = 0;
+    }, delay);
+  };
+}
```

Comment:
You can move it outside. If you want, you can describe hook useDebounce. Think it's interesting.